Marlin/src/feature/tmc_util.cpp

```diff
@@ -995,6 +999,45 @@
   }
   void tmc_disable_stallguard(TMC2660Stepper, const bool) {};
 
+  #if ENABLED(SPI_ENDSTOPS)
+
+    bool Endstops::tmc_spi_homing_check() {
+      bool hit = false;
+      #if X_SPI_SENSORLESS
+        if (TEST(tmc_spi_homing, X_AXIS) && stepperX.test_stall_status()) {
+          SBI(live_state, X_STOP);
+          hit = true;
+        }
+      #endif
```

Comment:
So, I made a couple of helpful changes here. First, `tmc_spi_homing` is now a bit mask, saving a couple bytes. Next, I changed the logic here so the endstop state is "sticky." That is, it only sets bits in `live_state` when the driver stalls, and doesn't clear them if the driver isn't stalled. This way a very short transient reading is guaranteed to give a positive result. (Of course, if the result is always checked when this is called, it makes no difference, but the code is a little smaller.) Finally, I applied your suggestion to return `true` if any of the endstops was triggered. It takes into account the possibility that some endstops are real ones.